### DISCUSSION TITLE
Enums can be object with custom error message

### DISCUSSION THREAD

**Discussion by akamfoad** [Automattic/mongoose#10116] [Ideas]
So today I had a problem with enums default failing message, I wanted to provide mine, but after searching whole docs found zero clue except using custom validators, but I felt that there should be something more easier syntax-wise, I found [https://stackoverflow.com/questions/37403907/mongoose-model-custom-error-message-for-enums](https://stackoverflow.com/questions/37403907/mongoose-model-custom-error-message-for-enums) question, which pointed out that you can provide an object to `enum` property like below:

```js
enum: {
  values: ["github", "twitter", "linkedin", "facebook"],
  message: "{VALUE} is not supported currently",
}
``` 

where `values` is an array of values and `message` is a string as any other validators like `unique`'s second string in the array.

I thought it would be wise to open an issue about that, but felt that it could something wider, that is why I'd liked to open this discussion.

So, is that something implicit? Like its doable because the nature of mongoose's schema?
If not, why there is no documentation about this?

**Comment by vkarpov15** [comment]
Added this to docs in https://github.com/Automattic/mongoose/commit/62ba2071586b357f48542ac2d3265ca0865f9c24 :+1:
